cpp/src/parquet/column_reader.cc

```diff
@@ -253,6 +253,13 @@ class SerializedPageReader : public PageReader {
 
   void set_max_page_header_size(uint32_t size) override { max_page_header_size_ = size; }
 
+  bool set_skip_page_callback(
+      std::function<bool(const format::PageHeader&)> skip_page_callback) override {
+    skip_page_callback_ = skip_page_callback;
+    has_skip_page_callback_ = true;
```

Comment:
We can avoid the has_skip_page_callback_ flag by simply checking `if (skip_page_callback_)`

https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool

Reply:
Done